libopenage/gamestate/game_main.cpp

```diff
@@ -37,17 +38,14 @@ GameMain::GameMain(const Generator &generator)
 }
 
 GameMain::~GameMain() {
-	for (auto &p : this->players) {
-		delete p;
-	}
 }
 
 unsigned int GameMain::player_count() const {
 	return this->players.size();
 }
 
 Player *GameMain::get_player(unsigned int player_id) {
-	return this->players.at(player_id);
+	return this->players.at(player_id).get();
```

Comment:
I tried to change this to ```shared_ptr<Player>``` locally - it does not seem to break anything.

Whay do you mean with not comfortable enough?